pkg/common/diskutil/file.go

```diff
@@ -1,16 +1,49 @@
 package diskutil
 
 import (
-	"io/ioutil"
 	"os"
+	"path/filepath"
 )
 
+// AtomicWriteFile writes data out.  It writes to a temp file first, fsyncs that file,
+// then swaps the file in.  os.Rename is an atomic operation, so this sequence avoids having
+// a partially written file at the final location.  Finally, fsync is called on the directory
+// to ensure the rename is persisted.
 func AtomicWriteFile(path string, data []byte, mode os.FileMode) error {
-	if err := ioutil.WriteFile(path+".tmp", data, mode); err != nil {
+	tmpPath := path + ".tmp"
+	if err := write(tmpPath, data, mode); err != nil {
 		return err
 	}
 
-	if err := os.Rename(path+".tmp", path); err != nil {
+	if err := os.Rename(tmpPath, path); err != nil {
+		return err
+	}
+
+	dir, err := os.Open(filepath.Dir(path))
+	if err != nil {
+		return err
+	}
+	defer dir.Close()
```

Comment:
We should check for the error coming back from Close(). Although it isn't likely to return an error since Sync() has been called, it seems defensive.